### ISSUE TITLE
rename files while saving to remove Unicode characters

### ISSUE THREAD

**Issue by yegor256** [objectionary/eo#1105]
We introduced a check for this situation in https://github.com/objectionary/eo/pull/1101 However, the problem is still there. If system encoding is not UTF-8, our compiler will break, because it won't be able to save files on disk. I suggest introducing a new object 'Disk' with two methods:

```
import org.cactoos.io.Input;
class Disk {
  static void save(Path, Input);
  static Input load(Path);
}
```

Then, redirect ALL read/write file operations to this class. Instead of using `java.nio.Files.write` we will use this `Disk` class.

**Comment by yegor256** [comment]
@Graur please, help

**Comment by yasamprom** [comment]
@yegor256 Why do we need `Disk`? May be we can just modify [`Save`](https://github.com/objectionary/eo/blob/master/eo-maven-plugin/src/main/java/org/eolang/maven/Save.java) ?

**Comment by yasamprom** [comment]
@yegor256 In current implementation we catch `IOException` in `Save`, but now we want to use `Input` and I guess we will need to use `Input.stream()` that throws `Exception`. It is bad style to catch `Exception`. So should we always rethrow it? Or is it problem of implementation of `Input`?

**Comment by yegor256** [comment]
@yasamprom you can use `org.cactoos.io.IoChecked` decorator inside `Save`

**Comment by Graur** [comment]
@yegor256 Can the `Disk` object behave like a global object?
e.g.
```
import org.cactoos.io.Input;
class Disk {
  private final Path path;
  private final Input in;

  static void save();
  static Input load();
}
```

**Comment by yegor256** [comment]
@yasamprom I didn't understand the question. How the attributes are used in your code and why they are there if all methods are static?

**Comment by Graur** [comment]
@yegor256 My bad, I forgot to remove them) But I think I got the idea.

**Comment by 0pdd** [comment]
@yegor256 3 puzzles [#1246](https://github.com/objectionary/eo/issues/1246), [#1247](https://github.com/objectionary/eo/issues/1247), [#1248](https://github.com/objectionary/eo/issues/1248) are still not solved.

**Comment by yasamprom** [comment]
@yegor256 let's close this issue. See PR objectionary/eo#1244

**Comment by yegor256** [comment]
@yasamprom thanks

**Comment by 0pdd** [comment]
@yegor256 3 puzzles [#1246](https://github.com/objectionary/eo/issues/1246), [#1248](https://github.com/objectionary/eo/issues/1248), [#1261](https://github.com/objectionary/eo/issues/1261) are still not solved; solved: [#1247](https://github.com/objectionary/eo/issues/1247).

**Comment by 0pdd** [comment]
@yegor256 2 puzzles [#1246](https://github.com/objectionary/eo/issues/1246), [#1292](https://github.com/objectionary/eo/issues/1292) are still not solved; solved: [#1247](https://github.com/objectionary/eo/issues/1247), [#1248](https://github.com/objectionary/eo/issues/1248), [#1261](https://github.com/objectionary/eo/issues/1261).

**Comment by 0pdd** [comment]
@yegor256 the puzzle [#1246](https://github.com/objectionary/eo/issues/1246) is still not solved; solved: [#1247](https://github.com/objectionary/eo/issues/1247), [#1248](https://github.com/objectionary/eo/issues/1248), [#1261](https://github.com/objectionary/eo/issues/1261), [#1292](https://github.com/objectionary/eo/issues/1292).

**Comment by 0pdd** [comment]
@yegor256 all 5 puzzles are solved here: [#1246](https://github.com/objectionary/eo/issues/1246), [#1247](https://github.com/objectionary/eo/issues/1247), [#1248](https://github.com/objectionary/eo/issues/1248), [#1261](https://github.com/objectionary/eo/issues/1261), [#1292](https://github.com/objectionary/eo/issues/1292).
